AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -839,26 +842,34 @@ - (ASCellNode *)supplementaryNodeOfKind:(NSString *)kind atIndexPath:(NSIndexPat
 - (id<ASSectionContext>)contextForSection:(NSInteger)section
 {
   ASDisplayNodeAssertMainThread();
-  ASDisplayNodeAssertTrue(section >= 0 && section < _sections.count);
-  return _sections[section].context;
+  return _pendingSections[section].context;
 }
 
 #pragma mark - elements dictionary
 
 //TODO Move this to somewhere else?
-+ (ASNodeContextTwoDimensionalDictionary *)deepImmutableCopyOfElementsDictionary:(ASNodeContextTwoDimensionalDictionary *)originalDict
++ (ASCollectionElementTwoDimensionalDictionary *)deepImmutableCopyOfElementsDictionary:(ASCollectionElementTwoDimensionalDictionary *)originalDict
+{
+  NSMutableDictionary *deepCopy = [NSMutableDictionary dictionaryWithCapacity:originalDict.count];
+  [originalDict enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull kind, ASCollectionElementTwoDimensionalArray * _Nonnull obj, BOOL * _Nonnull stop) {
+    deepCopy[kind] = [[NSArray alloc] initWithArray:obj copyItems:YES];
+  }];
+  return deepCopy;
```

Comment:
This is returning a mutable dictionary, is that intended or should it return `[deepCopy copy]`?